tests/std/tests/P1502R1_standard_library_header_units/custom_format.py

```diff
@@ -1,129 +1,221 @@
 # Copyright (c) Microsoft Corporation.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
+import json
 import os
+import re
 
 from stl.test.format import STLTestFormat, TestStep
 from stl.test.tests import TestType
 
 
+# Print noisy progress messages that are useful when working on this script.
+noisyProgress = False
+
+
+# P1502R1_standard_library_header_units/test.cpp cites the definition of "importable C++ library headers".
+def getImportableCxxLibraryHeaders():
+    return [
+        'algorithm',
+        'any',
+        'array',
+        'atomic',
+        'barrier',
+        'bit',
+        'bitset',
+        'charconv',
+        'chrono',
+        'codecvt',
+        'compare',
+        'complex',
+        'concepts',
+        'condition_variable',
+        'coroutine',
+        'deque',
+        'exception',
+        'execution',
+        'filesystem',
+        'format',
+        'forward_list',
+        'fstream',
+        'functional',
+        'future',
+        'initializer_list',
+        'iomanip',
+        'ios',
+        'iosfwd',
+        'iostream',
+        'istream',
+        'iterator',
+        'latch',
+        'limits',
+        'list',
+        'locale',
+        'map',
+        'memory',
+        'memory_resource',
+        'mutex',
+        'new',
+        'numbers',
+        'numeric',
+        'optional',
+        'ostream',
+        'queue',
+        'random',
+        'ranges',
+        'ratio',
+        'regex',
+        'scoped_allocator',
+        'semaphore',
+        'set',
+        'shared_mutex',
+        'source_location',
+        'span',
+        'spanstream',
+        'sstream',
+        'stack',
+        'stdexcept',
+        'stop_token',
+        'streambuf',
+        'string',
+        'string_view',
+        'strstream',
+        'syncstream',
+        'system_error',
+        'thread',
+        'tuple',
+        'type_traits',
+        'typeindex',
+        'typeinfo',
+        'unordered_map',
+        'unordered_set',
+        'utility',
+        'valarray',
+        'variant',
+        'vector',
+        'version',
+    ]
+
+
+def loadJsonWithComments(filename):
+    # This is far from a general-purpose solution (it doesn't attempt to handle block comments like /**/
+    # and comments appearing within strings like "cats // dogs"), but it's sufficient for header-units.json.
+    comment = re.compile('//.*')
+    with open(filename) as file:
+        replacedLines = [re.sub(comment, '', line) for line in file]
+        return json.loads(''.join(replacedLines))
+
+
+def getAllHeaders(headerUnitsJsonFilename):
+    buildAsHeaderUnits = loadJsonWithComments(headerUnitsJsonFilename)['BuildAsHeaderUnits']
+
+    # We want to build everything that's mentioned in header-units.json, plus all of the
+    # headers that were commented out for providing macros that control header inclusion.
+    return sorted(set(buildAsHeaderUnits + ['version', 'yvals.h', 'yvals_core.h']))

```

Comment:
we're already doing custom per-header stuff here, so again, we could keep the list of stuff to skip here (or in a separate json dict key) instead of using comments.